controllers/flinkcluster/suite_test.go

```diff
@@ -98,7 +100,11 @@ var _ = BeforeSuite(func() {
 
 var _ = AfterSuite(func() {
 	cancel()
+	ctx.Done()
 	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).ToNot(HaveOccurred())
+	timeout := 30 * time.Second
+	poll := 5 * time.Second
+	Eventually(func() error {
```

Comment:
kube-apiserver can take some time to be terminated when the test suite is being terminated.
[https://github.com/kubernetes-sigs/controller-runtime/issues/1571](Known issue)